sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryManager.scala

```diff
@@ -355,7 +355,7 @@ class StreamingQueryManager private[sql] (sparkSession: SparkSession) extends Lo
       // Make sure no other query with same id is active across all sessions
       val activeOption =
         Option(sparkSession.sharedState.activeStreamingQueries.putIfAbsent(query.id, this))
-      if (activeOption.isDefined) {
+      if (activeOption.isDefined || activeQueries.values.exists(_.id == query.id) {
```

Comment:
Just curious: are you seeing the actual case where activeOption is None but activeQueries contain such query? I'm not seeing the case, though I don't think adding this would hurt.